### DISCUSSION TITLE
RAMA_PREPRO_FILENAME

### DISCUSSION THREAD

**Discussion by MuhammadAttaElhamouly** [RosettaCommons/rosetta#145] [Rosetta]
Hi,

I have read about the RAMA_PREPRO_FILENAME line in the params file, but still need to understand more about how it works, Does it work for the energy function by using the probabilities of each backbone bin of phi and psi angles, or does it use go beyond that by limiting the backbone conformations for the high probabilities of the Ramachandran plot of each amino acid during designing, or it's more than this..

Also, I thought about using it in new residues to try giving it the same backbone behaviour of some amino acids after using RAMA_PREPRO_RESNAME and the three-letter name of the residue I want to select, is it possible and my expectations would work?

**Comment by NooraAz** [comment]
Hi,

> Does it work for the energy function by using the probabilities of each backbone bin of phi and psi angles ?

Yes. you can look up `database/scoring/score_functions/rama/fd/prepro.ramaProb` as an example. Also [Alford et al. (2017)](https://pubs.acs.org/doi/10.1021/acs.jctc.7b00125) explains it with details.


>  I thought about using it in new residues to try giving it the same backbone behaviour of some amino acids after using RAMA_PREPRO_RESNAME and the three-letter name of the residue I want to select

I'm not sure if I'm following, but if you mean instead of `prepro` you want for example `preala`, I don't think it is possible.

**Reply by MuhammadAttaElhamouly** [reply]
> Yes. you can look up `database/scoring/score_functions/rama/fd/prepro.ramaProb` as an example. Also [Alford et al. (2017)](https://pubs.acs.org/doi/10.1021/acs.jctc.7b00125) explains it with details.
> 
Thanks a lot, the article showing details about the concept and the energy terms that is using it. Is there more details about how the software handle in inside during the desining and other applications, limiting the backbone conformations for the high probabilities of the Ramachandran plot ?, like before even scoring, the sampling of the backbone angles, does it regulated to the allowed conformation only, to help ignoring low or impossible BB angles, or it sample all of it and score without ignoring these non allowed angles?


> I'm not sure if I'm following, but if you mean instead of `prepro` you want for example `preala`, I don't think it is possible.

I mean if I have NCAA like PHE drivatives, the Rama general case and even prepro can be theoritically applied for it, so is it possible to add for the params file of that PHE drivatives lines saying 

RAMA_PREPRO_FILENAME all.ramaProb prepro.ramaProb
RAMA_PREPRO_RESNAME PHE

If it's true that I can do that and the software will recognize it, then, is this the only modification I should do ?
